src/renderer/components/+namespaces/namespace-select.tsx

```diff
@@ -80,32 +73,61 @@ export class NamespaceSelect extends React.Component<Props> {
 
 @observer
 export class NamespaceSelectFilter extends React.Component {
-  render() {
-    const { contextNs, hasContext, toggleContext } = namespaceStore;
-    let placeholder = <>All namespaces</>;
+  @computed get placeholder(): React.ReactNode {
+    const namespaces = namespaceStore.getContextNamespaces();
+
+    switch (namespaces.length) {
+      case 1:
+        return <>Namespace: {namespaces[0]}</>;
+      case 2:
+        return <>Namespaces: {namespaces.join(", ")}</>;
+      default:
+        return "All Namespaces";
+    }
+  }
+
+  formatOptionLabel = ({ value: namespace, label }: SelectOption) => {
+    if (namespace) {
+      const isSelected = namespaceStore.hasContext(namespace);
+
+      return (
+        <div className="flex gaps align-center">
+          <FilterIcon type={FilterType.NAMESPACE}/>
+          <span>{namespace}</span>
+          {isSelected && <Icon small material="check" className="box right"/>}
+        </div>
+      );
+    }
+
+    return label;
+  };
 
-    if (contextNs.length == 1) placeholder = <>Namespace: {contextNs[0]}</>;
-    if (contextNs.length >= 2) placeholder = <>Namespaces: {contextNs.join(", ")}</>;
+  customizeOptions = (options: SelectOption[]): SelectOption[] => {
+    return [
+      { value: "", label: "All Namespaces" },
+      ...options
+    ];
+  };
 
+  onChange = ([{ value: namespace }]: SelectOption[]) => {
+    if (namespace) {
+      namespaceStore.toggleContext(namespace);
+    } else {
+      namespaceStore.toggleAll(); // "All namespaces" option clicked
+    }
+  };
+
+  render() {
     return (
       <NamespaceSelect
-        placeholder={placeholder}
+        isMulti={true}
```

Comment:
```suggestion
        isMulti
```

Reply:
Why was this resolved?

Reply:
How does this not go against the PR recommendations of not changing code style (especially without discussion).

And no I don't think I should be expecting code suggestions to just be silently ignored.